### PR TITLE
Сидорина Полина. Задача 1. Вариант 26. Проверка лексикографической упорядоченности двух строк.

### PR DESCRIPTION
**Pull request by PolinaSidorina**
Последовательная задача:
Сравнивается по два элемента каждой строки последовательно до того, пока не найдутся первые два разных символа. Затем определяется, какой из этих символов больше, и сравниваются строки целиком. Если первая строка больше - выводится 1, если вторая - 0.

Параллельная задача:
Каждая из двух строк делится на меньшие строки одинакового размера, количество таких строк зависит от числа процессов. Соответственно каждая строка отправляется в свой процесс, где происходит сравнение элементов по алгоритму последовательной задачи. На выходе процесса 0 - вторая строка больше, 1 - первая строка больше, 2 - строки одинаковые. Если процесс закончился не с выходом 2, то результат записывается как окончательный и сравнение заканчивается.

### CODE REVIEW

**Review by Sean0450 — APPROVED** [review]

File: tasks/mpi/sidorina_p_check_lexicographic_order/src/ops_mpi.cpp [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+  return true;
+}
+
+bool sidorina_p_check_lexicographic_order_mpi::TestMPITaskSequential::post_processing() {
+  internal_order_test();
+  reinterpret_cast<int*>(taskData->outputs[0])[0] = res;
+  return true;
+}
+
+bool sidorina_p_check_lexicographic_order_mpi::TestMPITaskParallel::pre_processing() {
```
Comment (allnes) [inline tasks/mpi/sidorina_p_check_lexicographic_order/src/ops_mpi.cpp:46]: please use mpi functionality in run()

File: tasks/mpi/sidorina_p_check_lexicographic_order/func_tests/main.cpp [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+    testMpiTaskSequential.run();
+    testMpiTaskSequential.post_processing();
+
+    ASSERT_EQ(ref_res[0], res[0]);
+    ASSERT_EQ(0, res[0]);
+  }
+}
+TEST(sidorina_p_check_lexicographic_order_mpi, Test_difference_4_element_1) {
+  boost::mpi::communicator world;
+  std::vector<std::vector<char>> str_ = {{'e', 'f', 'g', 'z'}, {'e', 'f', 'g', 'p'}};
```
Comment (aobolensk) [inline tasks/mpi/sidorina_p_check_lexicographic_order/func_tests/main.cpp:272]: All tests are using string of the same length, no tests for corner cases

**Review by aobolensk — CHANGES_REQUESTED** [review]

**Review by aobolensk — APPROVED** [review]

**Review by allnes — APPROVED** [review]
